packages/network-enablement-controller/src/NetworkEnablementController.ts

```diff
@@ -0,0 +1,353 @@
+import { BaseController } from '@metamask/base-controller';
+import type {
+  ControllerGetStateAction,
+  ControllerStateChangeEvent,
+  RestrictedMessenger,
+} from '@metamask/base-controller';
+import { BuiltInNetworkName, ChainId, toHex } from '@metamask/controller-utils';
+import { SolScope } from '@metamask/keyring-api';
+import type { MultichainNetworkControllerGetStateAction } from '@metamask/multichain-network-controller';
+import { toEvmCaipChainId } from '@metamask/multichain-network-controller';
+import type {
+  NetworkControllerGetStateAction,
+  NetworkControllerNetworkAddedEvent,
+  NetworkControllerNetworkRemovedEvent,
+  NetworkControllerStateChangeEvent,
+} from '@metamask/network-controller';
+import type { CaipChainId, CaipNamespace, Hex } from '@metamask/utils';
+import {
+  isHexString,
+  KnownCaipNamespace,
+  parseCaipChainId,
+} from '@metamask/utils';
+
+import { POPULATE_NETWORKS } from './constant';
+
+// Unique name for the controller
+const controllerName = 'NetworkEnablementController';
+
+/**
+ * Information about an ordered network.
+ */
+export type NetworksInfo = {
+  networkId: CaipChainId; // The network's chain id
+};
+
+type EnabledMap = Record<CaipNamespace, Record<string, boolean>>;
```

Comment:
What are your thoughts on providing JSDoc for this type? What does it represent?